feature/gnoi/os/tests/osinstall/osinstall_test.go

```diff
@@ -87,20 +88,33 @@ func TestOSInstall(t *testing.T) {
 	}
 	tc.fetchStandbySupervisorStatus(ctx, t)
 	tc.transferOS(ctx, t, false)
-	tc.activateOS(ctx, t, false)
-	if tc.dualSup {
+
+	if *deviations.OSActiavteRequiresReboot {
+		tc.activateOS(ctx, t, false /*standby*/, false /*noreboot*/)
+	} else {
+		tc.activateOS(ctx, t, false /*standby*/, true /*noreboot*/)
+	}
+
+	if !*deviations.NoOSInstallForStandbyRP && tc.dualSup {
 		tc.transferOS(ctx, t, true)
```

Comment:
Do you still need transferOS to backup in the case of `NoOSInstallForStandbyRP`?

Reply:
no, there is no need to trasnfer

Reply:
That's my point.  You still have `tc.transferOS(ctx, t, true)` in your PR, or I read it wrong?

Reply:
yes, I have it, but it is guarded by  !*deviations.NoOSInstallForStandbyRP.  This code block will be skipped when NoOSInstallForStandbyRP is set to true.

Reply:
ah, sorry I missed the double negative. Still it's a bit hard to read. Say if a dual SUP chassis requires reboot in activation, then you line#93 will reboot the whole chassis immediately, right? which means your line#98 logic will reach to a offline chassis?  Anyway you can help refactor it to make it more readable?

Reply:
sure, I will do that

Reply:
It seems I also missed the double negative regarding the noreboot flag and I may need a few more days to address this.

Reply:
@xw-g  thanks for the patience, I have addressed your comments.